### ISSUE TITLE
With new poetry installation (starting 02/28/2020) poetry tends to ignore private PyPi repos

### ISSUE THREAD

**Issue by apryiomka** [python-poetry/poetry#2120]
I am running poetry in docker container and have images the were built a few days ago vs the latest one. Nothing really changed other than we install poetry as part of the Docker build. Here is the diff of the installation that works vs the one that doesn't (poetry 1.0.3):
```

poetry==1.0.3							poetry==1.0.3
  - cachecontrol [required: >=0.12.4,<0.13.0, installed: 0.12	  - cachecontrol [required: >=0.12.4,<0.13.0, installed: 0.12
    - msgpack [required: >=0.5.2, installed: 1.0.0]	      |	    - msgpack [required: >=0.5.2, installed: 0.6.2]
    - requests [required: Any, installed: 2.23.0]	      |	    - requests [required: Any, installed: 2.22.0]
      - certifi [required: >=2017.4.17, installed: 2019.11.28	      - certifi [required: >=2017.4.17, installed: 2019.11.28
      - chardet [required: >=3.0.2,<4, installed: 3.0.4]      |	      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<3, installed: 2.9]	      |	      - idna [required: >=2.5,<2.9, installed: 2.8]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, 	      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0,
  - cachy [required: >=0.3.0,<0.4.0, installed: 0.3.0]		  - cachy [required: >=0.3.0,<0.4.0, installed: 0.3.0]
  - cleo [required: >=0.7.6,<0.8.0, installed: 0.7.6]		  - cleo [required: >=0.7.6,<0.8.0, installed: 0.7.6]
    - clikit [required: >=0.4.0,<0.5.0, installed: 0.4.2]     |	    - clikit [required: >=0.4.0,<0.5.0, installed: 0.4.1]
      - pastel [required: >=0.2.0,<0.3.0, installed: 0.2.0]   |	      - pastel [required: >=0.1.0,<0.2.0, installed: 0.1.1]
      - pylev [required: >=1.3,<2.0, installed: 1.3.0]		      - pylev [required: >=1.3,<2.0, installed: 1.3.0]
  - clikit [required: >=0.4.1,<0.5.0, installed: 0.4.2]	      |	  - clikit [required: >=0.4.1,<0.5.0, installed: 0.4.1]
    - pastel [required: >=0.2.0,<0.3.0, installed: 0.2.0]     |	    - pastel [required: >=0.1.0,<0.2.0, installed: 0.1.1]
    - pylev [required: >=1.3,<2.0, installed: 1.3.0]		    - pylev [required: >=1.3,<2.0, installed: 1.3.0]
  - html5lib [required: >=1.0,<2.0, installed: 1.0.1]		  - html5lib [required: >=1.0,<2.0, installed: 1.0.1]
    - six [required: >=1.9, installed: 1.14.0]			    - six [required: >=1.9, installed: 1.14.0]
    - webencodings [required: Any, installed: 0.5.1]		    - webencodings [required: Any, installed: 0.5.1]
  - importlib-metadata [required: >=1.1.3,<1.2.0, installed:  |	  - importlib-metadata [required: >=1.1.3,<1.2.0, installed:
    - zipp [required: >=0.5, installed: 3.0.0]		      |	    - zipp [required: >=0.5, installed: 2.2.0]
  - jsonschema [required: >=3.1,<4.0, installed: 3.2.0]		  - jsonschema [required: >=3.1,<4.0, installed: 3.2.0]
    - attrs [required: >=17.4.0, installed: 19.3.0]		    - attrs [required: >=17.4.0, installed: 19.3.0]
    - importlib-metadata [required: Any, installed: 1.1.3]    |	    - importlib-metadata [required: Any, installed: 1.5.0]
      - zipp [required: >=0.5, installed: 3.0.0]	      |	      - zipp [required: >=0.5, installed: 2.2.0]
    - pyrsistent [required: >=0.14.0, installed: 0.14.11]     |	    - pyrsistent [required: >=0.14.0, installed: 0.15.7]
      - six [required: Any, installed: 1.14.0]			      - six [required: Any, installed: 1.14.0]
    - setuptools [required: Any, installed: 41.0.1]		    - setuptools [required: Any, installed: 41.0.1]
    - six [required: >=1.11.0, installed: 1.14.0]		    - six [required: >=1.11.0, installed: 1.14.0]
  - keyring [required: >=20.0.1,<21.0.0, installed: 20.0.1]	  - keyring [required: >=20.0.1,<21.0.0, installed: 20.0.1]
    - importlib-metadata [required: Any, installed: 1.1.3]    |	    - importlib-metadata [required: Any, installed: 1.5.0]
      - zipp [required: >=0.5, installed: 3.0.0]	      |	      - zipp [required: >=0.5, installed: 2.2.0]
    - secretstorage [required: Any, installed: 3.1.2]		    - secretstorage [required: Any, installed: 3.1.2]
      - cryptography [required: Any, installed: 2.8]		      - cryptography [required: Any, installed: 2.8]
        - cffi [required: >=1.8,!=1.11.3, installed: 1.14.0]	        - cffi [required: >=1.8,!=1.11.3, installed: 1.14.0]
          - pycparser [required: Any, installed: 2.19]		          - pycparser [required: Any, installed: 2.19]
        - six [required: >=1.4.1, installed: 1.14.0]		        - six [required: >=1.4.1, installed: 1.14.0]
      - jeepney [required: >=0.4.2, installed: 0.4.2]		      - jeepney [required: >=0.4.2, installed: 0.4.2]
  - pexpect [required: >=4.7.0,<5.0.0, installed: 4.8.0]	  - pexpect [required: >=4.7.0,<5.0.0, installed: 4.8.0]
    - ptyprocess [required: >=0.5, installed: 0.6.0]		    - ptyprocess [required: >=0.5, installed: 0.6.0]
  - pkginfo [required: >=1.4,<2.0, installed: 1.5.0.1]		  - pkginfo [required: >=1.4,<2.0, installed: 1.5.0.1]
  - pyparsing [required: >=2.2,<3.0, installed: 2.4.6]		  - pyparsing [required: >=2.2,<3.0, installed: 2.4.6]
  - pyrsistent [required: >=0.14.2,<0.15.0, installed: 0.14.1 |	  - pyrsistent [required: >=0.14.2,<0.15.0, installed: 0.15.7
    - six [required: Any, installed: 1.14.0]			    - six [required: Any, installed: 1.14.0]
  - requests [required: >=2.18,<3.0, installed: 2.23.0]	      |	  - requests [required: >=2.18,<3.0, installed: 2.22.0]
    - certifi [required: >=2017.4.17, installed: 2019.11.28]	    - certifi [required: >=2017.4.17, installed: 2019.11.28]
    - chardet [required: >=3.0.2,<4, installed: 3.0.4]	      |	    - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
    - idna [required: >=2.5,<3, installed: 2.9]		      |	    - idna [required: >=2.5,<2.9, installed: 2.8]
    - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, in	    - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, in
  - requests-toolbelt [required: >=0.8.0,<0.9.0, installed: 0	  - requests-toolbelt [required: >=0.8.0,<0.9.0, installed: 0
    - requests [required: >=2.0.1,<3.0.0, installed: 2.23.0]  |	    - requests [required: >=2.0.1,<3.0.0, installed: 2.22.0]
      - certifi [required: >=2017.4.17, installed: 2019.11.28	      - certifi [required: >=2017.4.17, installed: 2019.11.28
      - chardet [required: >=3.0.2,<4, installed: 3.0.4]      |	      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<3, installed: 2.9]	      |	      - idna [required: >=2.5,<2.9, installed: 2.8]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, 	      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0,
  - shellingham [required: >=1.1,<2.0, installed: 1.3.2]	  - shellingham [required: >=1.1,<2.0, installed: 1.3.2]
```
I am not sure which package cause the issue, but the one on the right works while the one on the left doesn't.

Below is the pyrpoject.toml:
```
[[tool.poetry.source]]
name = "private-pypi"
url = "https://blah-blah-blah/simple/"

[tool.poetry]
name = "my-project"
version = "1.2.3"
description = "Base RESTfule service for online scoring"
authors = ["Alex Pryiomka <alex@example.com>"]
```

Now when i run `poetry add custom-package`, i get an error with the latest installation:
```
[ValueError]
Could not find a matching version of package custom-package
```

This works with the right image, but not the one on the left. **Also installation with pip works just fine in both images.**

**Comment by abn** [comment]
@apryiomka out of curiosity, within the same container does something like this work?

```dockerfile
FROM your/base/image

ENV PIP_EXTRA_INDEX=https://blah-blah-blah/simple/

RUN pip install custom-package
```

It could help rule out any issues introduced by a change in the base container. A change in the base container could show up as above (eg: abi/platform/version changes).

**Comment by apryiomka** [comment]
@abn , PIP works just fine with extra index. No changes were introduced, this is exactly the same installation except on was done before Friday last week and since Friday poetry seems to no recognize priviate PyPi anymore. The installation doesn't work for both new packages and the ones installed from the lock file. For the lock file i though get slightly different error:

**Comment by apryiomka** [comment]
This is an example of the project's dockerfile:
```
FROM blah-blah
RUN pip install -U pip poetry
ENV POETRY_VIRTUALENVS_CREATE=False
COPY pyproject.toml poetry.lock .flake8 ./

RUN poetry install --no-interaction
```

**Comment by timothyjlaurent** [comment]
@apryiomka  This was discovered last week, in #2109 and #2110. 

It was iltimately attributable to tomlkit. It can be fixed by installing 1.0.5

**Comment by apryiomka** [comment]
@timothyjlaurent thanks, missed the other two. It seems like a duplicate, i will close it now.
